### PR TITLE
Fix unbound var in search view

### PR DESCRIPTION
**Pull request by stsewd**
search can end up unbound if there isn't a valid facet

Fix https://sentry.io/organizations/read-the-docs/issues/1054512234/?project=148442&query=is%3Aunresolved

### CODE REVIEW

File: readthedocs/search/views.py [hunk 1]
```diff
@@ -105,7 +100,7 @@ def elastic_search(request, project_slug=None):
 
     # Make sure our selected facets are displayed even when they return 0 results
     for avail_facet in ALL_FACETS:
-        value = getattr(user_input, avail_facet)
+        value = getattr(user_input, avail_facet, None)
```
Comment (stsewd) [inline readthedocs/search/views.py]: if a default isn't given to `getattr` it raises an exception.

File: readthedocs/search/views.py [hunk 2]
```diff
@@ -56,8 +59,17 @@ def elastic_search(request, project_slug=None):
...
+        {
+            'project': ProjectSearch,
+            'domain': DomainSearch,
+            'file': PageSearch,
+            'all': AllSearch,
+        }
+    )
 
-    results = ''
+    results = None
```
Comment (stsewd) [inline readthedocs/search/views.py]: This feels more like an object than an empty string

File: readthedocs/search/views.py [hunk 3]
```diff
@@ -68,26 +80,9 @@ def elastic_search(request, project_slug=None):
...
-                query=user_input.query, user=request.user, **kwargs
-            )
-
-        elif user_input.type == 'all':
-            search = AllSearch(
-                query=user_input.query, user=request.user, **kwargs
-            )
-
+        search = search_facets[user_input.type](
+            query=user_input.query, user=request.user, **kwargs
```
Comment (humitos) [inline readthedocs/search/views.py]: The solution with defaultdict is more readable to me, I like it. Although, there is a new logic hidden: if the type does not exist (for any reason) we are going to return `ProjectSearch` results. Is that OK? would be better to just return no results at all?

Comment (humitos) [reply readthedocs/search/views.py]: We may want to add another condition next to the `user_input.query`

    ... and (user_input.type in search_facets):

and remove the lambda from the defaultdict.

Comment (stsewd) [reply readthedocs/search/views.py]: `project` is the default https://github.com/rtfd/readthedocs.org/blob/ccde7c598e2009e0fdb17505cd68c36153836376/readthedocs/search/views.py#L54

Comment (humitos) [reply readthedocs/search/views.py]: :+1: 

It seems we have the default set in multiples places then. I think we could just have only one.

**Pull request merged**